File: AssistantPlus.Windows/Utility/CommandLineWriter.cs
#pragma warning disable 0,
// ReSharper disable

using AssistantPlus;

namespace AssistantPlus.Utility {

	public class CommandLineWriter {

		#region structor

		private List<String> mView;

		// ----------------

		public CommandLineWriter (
		) {
			this.mView = [];
		}

		#endregion

		#region query

		public List<String> Done (
		) {
			return this.mView;
		}

		public Boolean Check (
			String  name,
			Boolean state = true
		) {
			if (state) {
				this.NextString(name);
			}
			return state;
		}

		// ----------------

		public void NextBoolean (
			Boolean value
		) {
			this.NextString(value.SelfLet((it) => (ConvertHelper.MakeBooleanToString(it))));
			return;
		}

		public void NextInteger (
			Integer value
		) {
			this.NextString(value.SelfLet((it) => (ConvertHelper.MakeIntegerToString(it, false))));
			return;
		}

		public void NextFloater (
			Floater value
		) {
			this.NextString(value.SelfLet((it) => (ConvertHelper.MakeFloaterToString(it, false))));
			return;
		}

		public void NextString (
			String value
		) {
			this.mView.Add(value);
			return;
		}

		public void NextStringList (
			List<String> value
		) {
			this.mView.AddRange(value);
			return;
		}

		#endregion

	}

}

File: Assistant/lib/utility/command_line_writer.dart
import '/common.dart';
import '/utility/convert_helper.dart';

// ----------------

class CommandLineWriter {

  // #region structor

  List<String> _view;

  // ----------------

  CommandLineWriter(
  ) :
    _view = [];

  // #endregion

  // #region query

  List<String> done(
  ) {
    return this._view;
  }

  Boolean check(
    String name, {
    Boolean state = true,
  }) {
    if (state) {
      this.nextString(name);
    }
    return state;
  }

  // ----------------

  Void nextBoolean(
    Boolean value,
  ) {
    return this.nextString(value.selfLet((it) => ConvertHelper.makeBooleanToString(it)));
  }

  Void nextInteger(
    Integer value,
  ) {
    return this.nextString(value.selfLet((it) => ConvertHelper.makeIntegerToString(it, false)));
  }

  Void nextFloater(
    Floater value,
  ) {
    return this.nextString(value.selfLet((it) => ConvertHelper.makeFloaterToString(it, false)));
  }

  Void nextString(
    String value,
  ) {
    this._view.add(value);
    return;
  }

  Void nextStringList(
    List<String> value,
  ) {
    this._view.addAll(value);
    return;
  }

  // #endregion

}
